src/less/skeleton/skeleton.less

```diff
@@ -0,0 +1,213 @@
+// using pseudo element to set as square
+.skeleton-image::after {
+    bottom: 0;
+    height: 100%;
+    left: 0;
+    position: absolute;
+    right: 0;
+    top: 0;
+    width: 100%;
+}
+
+span.skeleton-avatar,
+span.skeleton-button,
+span.skeleton-text,
+span.skeleton-image,
+span.skeleton-form {
+    display: inline-block;
+}
+
+span.skeleton-avatar + span,
+span.skeleton-button + span,
+span.skeleton-text + span,
+span.skeleton-image + span,
+span.skeleton-form + span {
+    margin-inline-start: var(--spacing-100);
+}
+
+div.skeleton-avatar,
+div.skeleton-button,
+div.skeleton-text,
+div.skeleton-image,
+div.skeleton-form {
+    margin-block-start: var(--spacing-150);
+}
+
+.skeleton-avatar--elevated,
+.skeleton-button--elevated,
+.skeleton-text--elevated,
+.skeleton-image--elevated,
+.skeleton-form--elevated {
+    background: var(--color-loading-fill-elevated);
+}
+
+@media screen and (max-width: @_screen-size-SM) {
+    .skeleton-avatar,
+    .skeleton-button,
+    .skeleton-text,
+    .skeleton-image,
+    .skeleton-form {
+        background: var(--color-loading-shimmer);
+    }
+    .skeleton-avatar--elevated,
+    .skeleton-button--elevated,
+    .skeleton-text--elevated,
+    .skeleton-image--elevated,
+    .skeleton-form--elevated {
+        background: var(--color-loading-shimmer-elevated);
+    }
+}
+
+.skeleton-avatar--purple,
+.skeleton-button--purple,
+.skeleton-text--purple,
+.skeleton-image--purple,
+.skeleton-form--purple {
+    background: var(--color-loading-ai-gradient-purple-subtle);
+}
+
+.skeleton-avatar--green,
+.skeleton-button--green,
+.skeleton-text--green,
+.skeleton-image--green,
+.skeleton-form--green {
+    background: var(--color-loading-ai-gradient-green-subtle);
+}
+
+.skeleton-avatar--blue,
+.skeleton-button--blue,
+.skeleton-text--blue,
+.skeleton-image--blue,
+.skeleton-form--blue {
+    background: var(--color-loading-ai-gradient-blue-subtle);
+}
+
+@media (prefers-reduced-motion: no-preference) {
+    @keyframes loading-frames {
+        0% {
+            background-position: 0% 0%;
+        }
+        50% {
+            background-position: 45% 0%;
+        }
+        100% {
+            background-position: 90% 0%;
+        }
+    }
+
+    @keyframes fade-in {
+        0% {
+            opacity: 0;
+        }
+        100% {
+            opacity: 1;
+        }
+    }
+
+    .skeleton-avatar,
+    .skeleton-button,
+    .skeleton-text,
+    .skeleton-image,
+    .skeleton-form {
+        /*  setting animation-iteration-count to 6 (5000/833ms)    */
+        /*  so it does not exceed 5s per WCAG 2.2.2    */
+        animation:
+            fade-in 500ms forwards,
+            loading-frames 833ms 667ms linear 6;
+        animation-fill-mode: both;
+        background-size: 200%;
+    }
+}
+
+// A progressive enhancement to make the skeleton image rounded
+@container skeleton-image (width > 79px) {
+    .skeleton-image {
```

Comment:
This won't work as .skeleton-image selector will not resolve on contiainer (because .skeleton-image is the root/container)

Reply:
Done, Fixed this by refactoring HTML.